File: Python/백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
node, edge, start=map(int, input().split())

graph = [[0]*(node+1) for _ in range(node+1)]
for i in range(edge):
    a, b=map(int,input().split())
    graph[a][b] = graph[b][a] = 1
visited1=[0]*(node+1) #node개수 + 1
visited2=[0]*(node+1)

def dfs(start):
    visited1[start]=1
    print(start, end=' ')
    for i in range(1,node+1):
        if graph[start][i]==1 and visited1[i]==0:
            dfs(i)
def bfs(start):
    queue=[start]
    visited2[start]=1
    while queue:
        start=queue.pop(0)
        print(start, end=' ')
        for i in range(1, node+1):
            if(visited2[i]==0 and graph[start][i]==1):
                queue.append(i)
                visited2[i]=1
dfs(start)
print()
bfs(start)
    

File: C99/백준/Silver/1260. DFS와 BFS/DFS와 BFS.c
#include<stdio.h>
#include <string.h>

int arr[1001][1001]={0,};
int visited[1001]={0,};
int node, edge, start;

int queue[1001];

void bfs(int start){
    int front=0;
    int rear=1;
    int pop;
    
    visited[start]=1;
    printf("%d ",start);
    queue[0]=start;
    while(front<rear){ //큐가 비어있으면 반복 종료
        pop=queue[front++]; //dequeue연산
        for(int i=1;i<=node+1;i++){
            if (arr[pop][i] && !visited[i]) {
				visited[i] = 1;
				printf("%d ", i);
				queue[rear++] = i; // enqueue 연산
			}
        }
    }
}

void dfs(int start){
    int i;
    visited[start]=1;
    printf("%d ",start);
    for (i=0;i<node+1;i++){
        if(arr[start][i]==1&&visited[i]==0){
            dfs(i);
        }
    }
    if (i == node) return;
}

int main(void){
    scanf("%d %d %d",&node, &edge, &start);
    
    for (int i=0;i<edge;i++){
        int a,b;
        scanf("%d %d",&a,&b);
        arr[a][b]=1;
        arr[b][a]=1;
    }
    dfs(start);
    printf("\n");
    memset(visited, 0, sizeof(visited));
    bfs(start);
}